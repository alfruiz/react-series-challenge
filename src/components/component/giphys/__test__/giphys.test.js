import React from 'react';
import { shallow } from 'enzyme';
import { spy } from "sinon";

import Giphys from "./../index";
import json from "./mock.json";



describe('<Giphys> ', () => {

  let wraper;

  beforeAll(() => {
    wraper = shallow(
      <Giphys />
    );
  });

  it('should be an empty div is not PROPS data passed!', () => {
    expect(wraper.children().length).toEqual(0);
  });

  it('should have number of children if data doenst have filter', () => {
    wraper.setProps({ data: {data: json.likes}, local: json  })
    expect(wraper.children().length).toEqual(json.likes.length);
  });

  it('if FILTER is set should only showed', () => {
    json.filter = json.likes[0].title.substring(1, 4);
    wraper.setProps({ data: {data: json.likes}, local: json  })
    expect(wraper.children().length).toEqual(1);
  });

  it('if FILTER is set should only showed', () => {
    json.filter = json.likes[0].title.substring(1, 4);
    wraper.setProps({ data: {data: json.likes}, local: json  })
    expect(wraper.children().length).toEqual(1);
  });

  it('if button is clicked should call handleKeep', () => {
    const handleKeepSpy = spy(wraper.instance(), "handleKeep");

    expect(handleKeepSpy.calledOnce).toBeFalsy();
    wraper.setProps({ data: { data: json.likes }, local: json , modifyLocal: () => {} });
    wraper.find('button').simulate('click', { preventDefault() { } });
    expect(handleKeepSpy.calledOnce).toBeTruthy();
  });

});