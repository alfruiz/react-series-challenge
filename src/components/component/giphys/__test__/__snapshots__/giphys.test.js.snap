// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Giphys>  Will check with the snaptshot  1`] = `
<div>
  <giphys__StyledGiphys
    key="3o7abttnfEEkYHspu8"
  >
    <giphys__StyledButton
      onClick={[Function]}
    >
      undo
    </giphys__StyledButton>
    <img
      alt="sad keith urban GIF by American Idol"
      src="https://media1.giphy.com/media/3o7abttnfEEkYHspu8/200w.webp"
    />
  </giphys__StyledGiphys>
</div>
`;
